C++.gitignore

```diff
@@ -30,3 +30,50 @@
 *.exe
 *.out
 *.app
+
+# CMake build directory
+build/
+
+# Visual Studio Code directory
+.vscode/
+
+# Sublime Text workspace files
+*.sublime-workspace
+
+# Eclipse directory
+.settings/
+
+# Code::Blocks directory
+bin/
+
+# Xcode directory
+*.xcodeproj/
+*.xcworkspace/
+
+# CMake build files
+CMakeFiles/
+CMakeCache.txt
+CMakeScripts/
+Testing/
+
+# VSCode directories and files
+.vscode/
+settings.json
```

Comment:
We usually have settings.json in the git repository.

Reply:
Removed